### PR TITLE
Fix routing with multiple routes contain 'path' params

### LINKED ISSUE #48: `404`s with parameter type `path`
Hi y'all,

we have just upgraded from `sanic==20.3.0` to `sanic==21.9.1` with `sanic-routing==0.7.1`.
Our test suite now shows 404s and 405s where we didn't expect them.

We run into this with Python 3.7, 3.8 and 3.9 on Windows and Ubuntu.

**Minimal Example**

```python
from sanic import Sanic
from sanic.response import text

app = Sanic("MyHelloWorldApp")


@app.get("/conversation/<conversation_id:path>/story")
async def story(request, conversation_id):
    return text("story")

@app.put("/conversation/<conversation_id:path>/tracker/events")
async def put_events(request, conversation_id):
    return text("put events")

@app.post("/conversation/<conversation_id:path>/tracker/events")
async def post_events(request, conversation_id):
    return text("post events")
```

**How to reproduce**
If you do a `curl http://127.0.0.1:8000/conversation/dasda-dasd/story`, this will result in a 404 (instead of the expected "story") If you comment out `put_events` or `post_events` it works fine.

I believe this is somewhat related to https://github.com/sanic-org/sanic/issues/2130 

Appreciate any help as this is currently blocking us to upgrade 🙌🏻

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed Changes**
- fix #48

### CODE REVIEW

**Comment by wochinge** [comment]
@ahopkins

File: tests/test_routing.py [hunk 1]
```diff
@@ -482,3 +482,25 @@ def handler2():
...
+@pytest.mark.parametrize("uri", ("a-random-path", "a/random/path"))
+def test_identical_path_routes_with_different_methods_similar_urls(uri):
+    def handler1():
+        return "handler1"
+
+    # test root level path with different methods
+    router = Router()
+    router.add("/constant/<foo:path>/story", handler1, methods=["GET", "OPTIONS"])
+    router.add("/constant/<foo:path>/tracker/events", handler1, methods=["PUT"])
+    router.add("/constant/<foo:path>/tracker/events", handler1, methods=["POST"])
```
Comment (ahopkins) [inline tests/test_routing.py:496]: Can we add `handler2` and `handler3` and then do asserts to make sure each is routed to the expected location? Should be an example to copy.

Comment (wochinge) [reply tests/test_routing.py:496]: will do

**Comment by ahopkins** [comment]
> @ahopkins 

Thanks for adding this. I just want to tweak the test slightly.

File: tests/test_routing.py (hunk 1 above)
Comment (wochinge) [reply tests/test_routing.py]: done (or should I assert the `route` as well?)

**Review by ahopkins — APPROVED** [review]
😎

File: tests/test_routing.py (hunk 1 above)
Comment (ahopkins) [reply tests/test_routing.py]: perfect

**Comment by ahopkins** [comment]
Linter is asking for annotation:

```
  sanic_routing/router.py:426: error: Need type annotation for "holder" (hint: "holder: List[<type>] = ...")
```

**Comment by wochinge** [comment]
fixing in a sec - just in a call atm

**Comment by ahopkins** [comment]
Will push to pypi tonight

**Pull request merged**